hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests/src/test/java/org/apache/hadoop/yarn/server/MiniYARNCluster.java

```diff
@@ -171,8 +173,11 @@ public MiniYARNCluster(
     this.numLocalDirs = numLocalDirs;
     this.numLogDirs = numLogDirs;
     this.enableAHS = enableAHS;
-    String testSubDir = testName.replace("$", "");
-    File targetWorkDir = new File("target", testSubDir);
+    String testSubDir = testName.replace("$", "").concat("-yarn");
```

Comment:
I wonder if this testSubDir should just be "yarn".   The targetWorkDirRoot already has the class name in it.

Reply:
I think even better it should be unique folder name "yarn_TESTXYZ" . This way, multiple test cases within the same class won't override the existing folder.

Reply:
I was thinking "yarn" because we create a "dfs" directory for hdfs.   We could make random subdirs under it, but I don't think it's necessary.